PPos[] pos
ControlInput[] in
float32 energy
int16 colnum
int16 num
int16 particle_id
